staging/vhost-device-sound/src/stream.rs

```diff
@@ -408,6 +411,29 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_display_fmt() {
+        let state = PCMState::Stop;
+        let expected = "VIRTIO_SND_R_PCM_STOP";
+        assert_eq!(format!("{}", state), expected);
+    }
+
+    #[test]
+    fn test_logging() {
+        env_logger::builder()
+            .filter_level(log::LevelFilter::Trace) // Set the log level to trace.
+            .is_test(true) // Enable test mode to capture log output.
+            .try_init()
+            .ok();
+
+        let data_descriptor = Descriptor::new(0, 0, 0, 0);
+        let msg = iomsg();
+        let message = Arc::new(msg);
+        let direction = Direction::Input;
+        let buffer = Buffer::new(data_descriptor, message, direction);
+        log::trace!("dropping {:?} buffer {:?}", direction, buffer);
+    }
```

Comment:
this test doesn't assert for failure but increases the code region coverage. so if the log::trace doesn't show up, the test will be useless.

Reply:
Instead of using the logger, can we use `format!`, I mean something like this:
```
assert_eq!(format!("{dropping:?} {direction:?}"), ".....");
```

Reply:
I'm not sure about that, plus I don't know how I'm supposed to use it in this case. because it will be 
```
assert_eq!(format!(....), format!(....))
```

Reply:
For the `direction` it could be `assert_eq!(format!("{direction:?}"), "Input");`, for the buffer is a bit difficult since we have a reference, but something like this should work:
```
        assert_eq!(
            format!("{buffer:?}"),
            format!(
                "Buffer {{ pos: 0, direction: Input, message: {:?} }}",
                &Arc::as_ptr(&buffer.message)
            )
        );
```

So at the end I'd suggest something like this:
```patch
diff --git a/staging/vhost-device-sound/src/stream.rs b/staging/vhost-device-sound/src/stream.rs
index 045e727..fde7b55 100644
--- a/staging/vhost-device-sound/src/stream.rs
+++ b/staging/vhost-device-sound/src/stream.rs
@@ -337,7 +337,6 @@ impl Drop for Buffer {
 mod tests {
     use std::fmt::Write;
 
-    use env_logger;
     use vhost_user_backend::{VringRwLock, VringT};
     use virtio_bindings::bindings::virtio_ring::{VRING_DESC_F_NEXT, VRING_DESC_F_WRITE};
     use virtio_queue::{mock::MockSplitQueue, Descriptor, Queue, QueueOwnedT};
@@ -420,18 +419,19 @@ mod tests {
 
     #[test]
     fn test_logging() {
-        env_logger::builder()
-            .filter_level(log::LevelFilter::Trace) // Set the log level to trace.
-            .is_test(true) // Enable test mode to capture log output.
-            .try_init()
-            .ok();
-
         let data_descriptor = Descriptor::new(0, 0, 0, 0);
         let msg = iomsg();
         let message = Arc::new(msg);
         let direction = Direction::Input;
         let buffer = Buffer::new(data_descriptor, message, direction);
-        log::trace!("dropping {:?} buffer {:?}", direction, buffer);
+        assert_eq!(format!("{direction:?}"), "Input");
+        assert_eq!(
+            format!("{buffer:?}"),
+            format!(
+                "Buffer {{ pos: 0, direction: Input, message: {:?} }}",
+                &Arc::as_ptr(&buffer.message)
+            )
+        );
     }
 
     #[test]
```

Reply:
ok thank you :)